src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetWidget.java

```diff
@@ -854,13 +861,20 @@ private void manageToolbarSizes()
       knitDocumentButton_.setText(width < 450 ? "" : knitCommandText_);
       
       if (editor_.getFileType().isRd() || editor_.getFileType().isJS() || 
-          editor_.getFileType().isSql() ||editor_.getFileType().canPreviewFromR())
+          editor_.getFileType().isSql())
       {
          String preview = editor_.getFileType().getPreviewButtonText();
          srcOnSaveLabel_.setText(width < 450 ? preview : preview + " on Save");
       }
+      else if (hasCustomSource())
+      {
+         srcOnSaveLabel_.setText(width < 450 ? "Function" : "Function on Save");
```

Comment:
This text seems odd to me -- why not just keep `Source on Save`?

Reply:
Yeah, we could, we change the `source` message, but I think I agree with you, might be better just to keep the usual `Source on Save`, will change.

Reply:
+1